src/gui/mzroll/tabledockwidget.cpp

```diff
@@ -1765,6 +1765,61 @@ void TableDockWidget::showIntegrationSettings()
   log.exec();
 }
 
+void TableDockWidget::moveSelectedRows(QString destinationTableName) {
+
+  QString sourceTableName = this->titlePeakTable->text();
+  QList<QPointer<TableDockWidget>> peaksTableList = _mainwindow->getPeakTableList();
+  QList<shared_ptr<PeakGroup>> selected = getSelectedGroups();
+  QString bookmarkTableName = _mainwindow->bookmarkedPeaks->titlePeakTable->text();
+
+  if(bookmarkTableName == destinationTableName) {
+    BookmarkTableDockWidget* destinationPeakTable = _mainwindow->bookmarkedPeaks;
+    for(auto group : selected) {
+      destinationPeakTable->addPeakGroup(group.get());
+    }
+    destinationPeakTable->showAllGroups();
+    destinationPeakTable->setVisible(true);
+  } else {
+    TableDockWidget* destinationPeakTable = nullptr;
+    for (auto table : peaksTableList) {
+        if (table->titlePeakTable->text() == destinationTableName) {
+            destinationPeakTable = table;
+        }
+        if(destinationPeakTable != nullptr) {
+          break;
+        }
+    }
+
+    if (!destinationPeakTable) return;
+    for(auto group : selected) {
+      destinationPeakTable->addPeakGroup(group.get());
+    }
+    destinationPeakTable->showAllGroups();
+  }
+
+  if(bookmarkTableName == sourceTableName) {
+    BookmarkTableDockWidget* sourcePeakTable = _mainwindow->bookmarkedPeaks;
+    for(auto group : selected) {
+      sourcePeakTable->deleteGroup(group.get());
+    }
```

Comment:
Since the items in this table are still "selected", I think it would be better if we use the `deleteSelectedItems` method for both Bookmark table as well as other Peak tables. The method takes care of many other things in addition to deleting groups.